src/modules/app-db/repositories/admin.repository.ts

```diff
@@ -460,15 +460,24 @@ export class AdminRepository {
     });
   }
 
-  async getNftDetailTrial(id: NftId) {
-    return this.nfts.findOne({
+  async getDetailTrial(id: NftId, collectionID: string, walletAddress: string) {
+    //List collections
+    const nft = await this.nfts.findOne({
       relations: {
         artwork: true,
       },

```

Comment:
we can fetch this in one query with relations { collection: { wallet : true } } and similarly adjusted where condition